spec/features/accessibility_spec.rb

```diff
@@ -0,0 +1,180 @@
+
+  def find_routes(type, resource, subresource)
+    routes = Rails.application.routes.routes.select do |route|
+      route.verb == "GET" && route.path.spec.to_s.start_with?("/#{type}")
+    end
+
+    routes.map do |route|
+      route_path = route.path.spec.to_s
+      route_path
+        .gsub("/#{type}s/:id", "/#{type}s/#{resource.id}")
+        .gsub(":#{type.underscore}_id", resource.id.to_s)
+        .gsub(":id", subresource.id.to_s)
+        .gsub("(.:format)", "")
+    end
+  end
+
+  before do
+    allow(user).to receive(:need_two_factor_authentication?).and_return(false)
+    sign_in(user)
+  end
+
+  context "when viewing user pages" do
+    let(:user_paths) do
+      Rails.application.routes.routes.select { |route| route.verb == "GET" && route.path.spec.to_s.start_with?("/user") }.map { |route|
+        route_path = route.path.spec.to_s
+        route_path.gsub(":id", other_user.id.to_s).gsub(":user_id", other_user.id.to_s).gsub("(.:format)", "")
+      }.uniq
+    end
+
+    it "is has accessible pages" do
+      user_paths.each do |path|
+        visit(path)
+        expect(page).to have_current_path(path)
+        expect(page).to be_axe_clean.according_to :wcag2aa
+      end
+    end
+  end
+
+  context "when viewing organisation pages" do
+    let(:parent_relationship) { create(:organisation_relationship, parent_organisation: other_user.organisation) }
+    let(:child_relationship) { create(:organisation_relationship, child_organisation: other_user.organisation) }
+    let(:organisation_paths) do
+      routes = find_routes("organisation", other_user.organisation, other_user.organisation).reject do |route|
+        route.match?(/\A\/organisations\/#{other_user.organisation_id}\z/) ||
+          route.include?("filters/update")
+      end
+      routes << "/organisations/#{other_user.organisation_id}/details"
+      route_mappings = {
+        "/schemes/csv-download" => "?download_type=combined",
+        "logs/csv-download" => "?codes_only=false&years[]=2024",
+        "filters/update" => "?codes_only=false",
+        "stock-owners/remove" => "?target_organisation_id=#{child_relationship.parent_organisation.id}",
+        "managing-agents/remove" => "?target_organisation_id=#{parent_relationship.child_organisation.id}",
+      }
+
+      routes.map do |route|
+        additional_params = route_mappings.find { |pattern, _| route.include?(pattern) }&.last
+        route += additional_params if additional_params
+        route
+      end
+    end
+
+    it "is has accessible pages" do
+      organisation_paths.each do |path|
+        visit(path)
+        expect(page).to have_current_path(path)
+        expect(page).to be_axe_clean.according_to :wcag2aa
+      end
+    end
+  end
+
+  context "when viewing lettings log pages" do
+    let(:bulk_upload) { create(:bulk_upload, user:) }
+    let(:lettings_log) { create(:lettings_log, :completed, assigned_to: other_user, bulk_upload_id: bulk_upload.id) }
```

Comment:
Presumably this will only end up checking pages relevant to this log - are there any types of question that would be missed because of that? (I think it's fine as long as we try each question type at least once)

Reply:
Yeah, that was my thinking too. I think we're hitting all of the types, I was mostly worried about select, but we've got org and user questions